owncloudApp/src/main/java/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -394,58 +414,44 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                 // account at this point
                 // since account setup can set only one account at time
                 setAccount(accounts[0]);
-                populateDirectoryList();
+                updateDirectoryList();
             }
         }
     }
 
-    private void populateDirectoryList() {
-        setContentView(R.layout.uploader_layout);
-        initToolbar();
-
-        mSortOptionsView = findViewById(R.id.options_layout);
-        if (mSortOptionsView != null) {
-            mSortOptionsView.setOnSortOptionsListener(this);
-            mSortOptionsView.setOnCreateFolderListener(this);
-            mSortOptionsView.selectAdditionalView(SortOptionsView.AdditionalView.CREATE_FOLDER);
-        }
-
-        ListView mListView = findViewById(android.R.id.list);
-        mEmptyListMessage = findViewById(R.id.empty_list_view);
+    private void updateDirectoryList() {
+        initToolbar(mParents.peek());
 
         String full_path = generatePath(mParents);
-
         Timber.d("Populating view with content of : %s", full_path);
-
         mFile = getStorageManager().getFileByPath(full_path);
         if (mFile != null) {
-            Vector<OCFile> files = getStorageManager().getFolderContent(mFile);
-            files = sortFileList(files);
-
-            mAdapter = new ReceiveExternalFilesAdapter(
-                    this, files, getStorageManager(), getAccount());
-            mListView.setAdapter(mAdapter);
+            if(mAdapter == null) {
+                mAdapter = new ReceiveExternalFilesAdapter(
+                        this, getStorageManager(), getAccount());
+                mListView.setAdapter(mAdapter);
+            }
```

Comment:
Just curiosity, when will `mAdapter` be null?

Reply:
When `onCreate()` is called we can not yet set the adapter as this would lead the application to crash if thumbnails need to be rendered. So the Adapter is created lazily. However we do not want to override the adapter but only create it once. Therefore the null check.